src/huggingface_hub/repository.py

```diff
@@ -1023,15 +1023,22 @@ def auto_track_binary_files(self, pattern: Optional[str] = ".") -> List[str]:
                 continue
 
             path_to_file = os.path.join(os.getcwd(), self.local_dir, filename)
-            is_binary = is_binary_file(path_to_file)
 
-            if (
-                is_binary
-                and not is_tracked_with_lfs(path_to_file)
-                and not is_git_ignored(path_to_file)
-            ):
-                self.lfs_track(filename)
-                files_to_be_tracked_with_lfs.append(filename)
+            if not (is_tracked_with_lfs(path_to_file) or is_git_ignored(path_to_file)):
```

Comment:
```suggestion
            if not (is_tracked_with_lfs(path_to_file) and not is_git_ignored(path_to_file)):
```

Reply:
Here the initial `not` is outside of the statement so it encompasses both; I'm happy to follow your proposal but the parenthesis around the two statements should be removed. Would you find that clearer?

(it's on me, I updated that in the latest commit so that it fits on the same line, but it seems to hinder understanding)

Reply:
oh my bad - I read wrongly